### ISSUE TITLE
Extend basic VAE example for regression models

### ISSUE THREAD

**Issue by nscherf** [neural-data-science-lab/intro_dgm#1]
We need a basic VAE example for regression models. So far, the class only implements binary and categorical classification. 

The simplest case would be Gaussian / Normal outputs (mean and variance) at each output node of the decoder. We also need to add the respective computation of the log_prob for this case (should essentially be MSE if we use the mean of the output .

**Comment by nscherf** [comment]
vaes/vae_example.ipynb

**Comment by nscherf** [comment]
Thanks @Kajalsingla1997 for implementing the normal distribution cbf6648d2d64dbdf58a5d8c187570406b57bb7b7

**Comment by nscherf** [comment]
Still open:

- [x] implement log_variance estimate in the decoder (output mean and log_variance)
- [x] use estimated log_variance in the loss function
- [x] test variance loss
- [x] use estimated log_variance in the sampling method of the Decoder
- [x] test sampling

**Comment by nscherf** [comment]
Thus far, the samples generated by the heteroscedastic model (per sample variance) from MNIST are not convincing, though the model seems to have converged during training. 

According to Chapter 5 (Loss Functions) from https://udlbook.github.io/udlbook/ we could try squaring our variance estimate instead of predicting the log variance (though that shouldn't be an issue). 

- [ ] check out the implementation of a full gaussian loss function at https://github.com/udlbook/udlbook/blob/main/Notebooks/Chap05/5_1_Least_Squares_Loss.ipynb
- [ ] implement square(-root) estimate of variance
